### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by andriy-shymkiv**

### CODE REVIEW

File: src/transformState.js [hunk 1]
```diff
@@ -5,7 +5,23 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
+    if (obj.type === 'addProperties') {
+      Object.assign(state, obj.extraData);
+    }
```
Comment (referee1988) [inline src/transformState.js:11]: Checklist 2. [CODE STYLE]: Use switch statement if you have limited amount of conditions.

Replace if with switch

File: src/transformState.js [hunk 2]
```diff
@@ -5,7 +5,23 @@
...
+
+    if (obj.type === 'removeProperties') {
+      for (const key of obj.keysToRemove) {
+        delete state[key];
+      }
+    }
```
Comment (referee1988) [inline src/transformState.js:17]: as above

File: src/transformState.js [hunk 3]
```diff
@@ -5,7 +5,23 @@
...
+
+    if (obj.type === 'clear') {
+      for (const key in state) {
+        delete state[key];
+      }
+    }
```
Comment (referee1988) [inline src/transformState.js:23]: as above

**Review by VikNikolaiev — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 4]
```diff
@@ -5,7 +5,42 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const obj of actions) {
```
Comment (mykola-koval) [inline src/transformState.js:8]:
```suggestion
  for (const action of actions) {
```

File: src/transformState.js [hunk 5]
```diff
@@ -5,7 +5,42 @@
...
+//   for (const key of obj.keysToRemove) {
+//     delete state[key];
+//   }
+// }
+
+// if (obj.type === 'clear') {
+//   for (const key in state) {
+//     delete state[key];
+//   }
+// }
```
Comment (mykola-koval) [inline src/transformState.js:46]: remove comments

**Review by mkyloa — CHANGES_REQUESTED** [review]

File: src/transformState.js [hunk 6]
```diff
@@ -5,7 +5,26 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+
```
Comment (mykola-koval) [inline src/transformState.js:12]:
```suggestion
```

**Review by mkyloa — APPROVED** [review]

**Pull request closed without merging**